src/parse_json_for_plots.py

```diff
@@ -0,0 +1,241 @@
+    x_label = group.capitalize() + " Time (s)"
+    plt.clf()
+    plt.figure(figsize=(10, 6))
+
+    # Add labels and title
+    plt.xlabel(x_label)
+    plt.ylabel("Compression Ratio")
+    plt.title(f"Compression Ratio vs {x_label} for Different Compression Types")
+
+    sns.boxplot(
+        x="write_time",
+        y="compression_ratio",
+        hue="compression_type",
+        data=df,
+        palette=palette_dict,
+    )
+
+    save_plot_as_png(
+        plt,
+        f"data/json/{group}_errorbars_{Path(path_to_file).stem}.png",
+    )
+
+
+def plot_relplot_subplots_benchmarks(
+    data: pd.DataFrame,
+    group: str,
+    x_axis: str,
+    y_axis: str,
+    hue: str,
+    path_to_file=None,
+):
+    graph = sns.relplot(
+        data=data,
+        x=x_axis,
+        y=y_axis,
+        col=hue,
+        hue=hue,
+        facet_kws=dict(sharex=False, sharey=False),
+    )
+    graph.set_axis_labels(x_axis.capitalize(), f"Mean {group} time (s)")
+
+    if path_to_file is not None:
+        save_plot_as_png(
+            graph,
+            f"data/json/test_facet_kws{group}_relplot_{Path(path_to_file).stem}.png",
+        )
+
+
+def plot_relplot_benchmarks(
+    data: pd.DataFrame,
+    group: str,
+    x_axis: str,
+    y_axis: str,
+    hue: str,
+    size: str,
+    title: str,
+    path_to_file=None,
+):
+    graph = sns.relplot(
+        data=data,
+        x=x_axis,
+        y=y_axis,
+        hue=hue,
+        style=hue,
+        size=size,
+        col=title,
+        height=4,
+        aspect=1.5,
+    )
+    graph.set_axis_labels(f"Mean {group} time (s)", y_axis.capitalize())
+
+    if path_to_file is not None:
+        save_plot_as_png(
+            graph,
+            f"data/json/test_{group}_relplot_{Path(path_to_file).stem}.png",
+        )
+
+
+def save_plot_as_png(plt: plt, path_to_file: str) -> None:
+    plt.savefig(path_to_file, format="png", dpi=300)
+
+
+if __name__ == "__main__":
+    zarr_v2_path = "data/json/0007_zarr-python-v2.json"
+    zarr_v3_path = "data/json/0008_zarr-python-v3.json"
+    tensorstore_path = "data/json/0009_tensorstore.json"
+
+    benchmarks_df = get_benchmarks_dataframe(
+        (zarr_v2_path, zarr_v3_path, tensorstore_path),
+        package_ids=("zarr_python_2", "zarr_python_3", "tensorstore"),
+    )
+    benchmarks_zarr_v2 = benchmarks_df[benchmarks_df.package == "zarr_python_2"]
+    write_zarr_v2 = benchmarks_zarr_v2[benchmarks_zarr_v2.group == "write"]
+    read_zarr_v2 = benchmarks_zarr_v2[benchmarks_zarr_v2.group == "read"]
+
+    write_zarr_v2_chunks_200 = write_zarr_v2[write_zarr_v2.chunk_size == 200]
+    read_zarr_v2_chunks_200 = read_zarr_v2[read_zarr_v2.chunk_size == 200]
+
+    plot_relplot_benchmarks(
+        write_zarr_v2_chunks_200,
+        "write",
+        "stats.mean",
+        "compression_ratio",
+        "compressor",
+        "compression_level",
+        "package",
+        zarr_v2_path,
+    )
```

Comment:
Maybe use longform refs here, as there's a lot of parameters! I had to scroll up and down a lot to figure out what was what 😅 
```
plot_relplot_benchmarks(
        write_zarr_v2_chunks_200,
        group="write",
        x_axis="stats.mean",
        y_axis="compression_ratio",
        hue="compressor",
        size="compression_level",
        title="package",
        path_to_file=zarr_v2_path,
    )
```
If you wanted, you could enforce this with `*` - see [David's comments on one of my other PRs](https://github.com/HEFTIEProject/zarr-benchmarks/pull/22#discussion_r2023067133)